typescript/packages/plugins/ionic/src/ionic.service.ts

```diff
@@ -0,0 +1,169 @@
+import { Tool } from "@goat-sdk/core";
+import { EVMTransactionOptions, EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { z } from "zod";
+import { parseUnits, formatUnits, type Address } from "viem";
+import type { HealthMetrics, PoolData } from "./types";
+import { ionicProtocolAddresses } from "./ionic.config";
+
+import { poolLensAbi } from "./abis/PoolLens";
+import { comptrollerAbi } from "./abis/Comptroller"; 
+import { poolAbi } from "./abis/Pool";
+import { poolDirectoryAbi } from "./abis/PoolDirectory";
+
+import {
+  SupplyAssetParameters,
+  BorrowAssetParameters,
+  GetHealthMetricsParameters
+} from "./parameters";
+
+export class IonicService {
+  private supportedTokens: string[];
+
+  constructor(supportedTokens?: string[]) {
+    this.supportedTokens = supportedTokens || ["USDC", "WETH"];
+  }
+
+  private async getAssetConfig(chainId: number, symbol: string): Promise<{ address: Address, decimals: number }> {
+    const config = ionicProtocolAddresses[chainId]?.assets?.[symbol];
+    if (!config?.address || config.decimals === undefined) {
+      throw new Error(`Asset ${symbol} not found in Ionic Protocol addresses for chain ${chainId}`);
+    }
+    return { address: config.address as Address, decimals: config.decimals };
+  }
+
+  @Tool({
+    name: "ionic_supply_asset",
+    description: "Supply an asset to an Ionic Protocol pool"
+  })
+  async supplyAsset(walletClient: EVMWalletClient, parameters: SupplyAssetParameters) {
+    const { poolId, asset, amount } = parameters;
+    const chainId = walletClient.getChain().id;
+
+    try {
+
+      const poolAddress = ionicProtocolAddresses[chainId]?.pools[poolId] as Address;
+      if (!poolAddress) {
+        throw new Error(`Pool address not found for pool ID ${poolId} on chain ${chainId}`);
+      }
+
+      console.log("poolAddress", poolAddress);
+
+      const assetConfig = await this.getAssetConfig(chainId, asset);
+      const amountBigInt = parseUnits(amount, assetConfig.decimals);
```

Comment:
The input is already in base units, this shouldn't be needed

Reply:
sure